app/assets/javascripts/initializers/initializeBaseUserData.js

```diff
@@ -110,9 +110,40 @@ function setCurrentUserToNavBar(user) {
   }
 }
 
+/**
+ * Responsible for hiding or showing elements that match each of the given user
+ * policies.  While this function is "oblivious" to what it's hiding, it
+ * coordinates between the rendered HTML and the user data to show or hide
+ * elements that present functionality available or not available to the given
+ * user.
+ *
+ * @param {Object} user with a policies properties that is an array of objects.
+ *        Each of those policy objects has a dom_class and forbidden property.
+ *
+ * A critical assumption is that we are not employing "security through
+ * obscurity".  That is to say, if we accidentally show the link, the server
+ * will enforce the correct policy.
+ */
+function applyPolicyToggles(user) {
+  if (user.policies) {
+    user.policies.forEach(function (policy) {
+      const elements = document.getElementsByClassName(policy.dom_class);
+      for (let i = 0; i < elements.length; i++) {
+        const element = elements[i];
+        if (policy.forbidden) {
+          element.classList.add('hidden');
+        } else {
+          element.classList.remove('hidden');
+        }
```

Comment:
I think we are going to need to flip the approach to "UI is hidden by default" - here's an example where I applied the `js-policy-article-create` to the notification button - it flashes up before disappearing since this operation is async. I think on balance the better user experience is to progressively enhance in the opposite direction, layering in the UI for the reserved features asynchronously.

https://user-images.githubusercontent.com/20773163/161236107-f8b76d6d-56f3-4a1f-ac2b-335c2164eea7.mp4

Reply:
I agree.  I started poking around the main "Create Post" button and I definitely don't want something going "I'm here and poof I'm gone."

Which has me considering whether the "dom_class_for" method should return a string with two classes, defaulting to "js-policy-article-create hidden" and if you pass `hidden: false` it would only have one.  In other words, favor the default but make space for the other.

Reply:
I went ahead and reworked the policy object to have `dom_class` and `visible` elements.  It's almost no longer about "authorization policies".  Which feels even better.

Also, I flipped this so things will flicker "onto" the page instead of flickering out.